packages/did-provider-ens/src/ens-did-provider.ts

```diff
@@ -0,0 +1,76 @@
+import { IIdentifier, IKey, IService, IAgentContext, IKeyManager } from '@veramo/core'
+import { AbstractIdentifierProvider } from '@veramo/did-manager'
+
+import Debug from 'debug'
+const debug = Debug('veramo:ens-did:identifier-provider')
+
+type IContext = IAgentContext<IKeyManager>
+
+/**
+ * {@link @veramo/did-manager#DIDManager} identifier provider for `did:ens` identifiers
+ * @public
+ */
+export class EnsDIDProvider extends AbstractIdentifierProvider {
+  private defaultKms: string
+
+  constructor(options: { defaultKms: string }) {
+    super()
+    this.defaultKms = options.defaultKms
+  }
+
+  async createIdentifier(
+    { kms, alias, options }: { kms?: string; alias?: string; options: any },
+    context: IContext,
+  ): Promise<Omit<IIdentifier, 'provider'>> {
+    const keyType = options?.keyType || 'Secp256k1'
+    const key = await context.agent.keyManagerCreate({ kms: kms || this.defaultKms, type: keyType })
+
+    const identifier: Omit<IIdentifier, 'provider'> = {
```

Comment:
In its current form, these identifiers wouldn't be usable for signing credentials, because the keys created here don't get reflected in the resolved DID document for the `did:ens`.
This pattern is currently being used by the `did-provider-web` package, but the assumption there is that the same agent managing the DID is the one that serves the DID document (using [@veramo/remote-server#WebDidDocRouter](https://github.com/uport-project/veramo/blob/fb22e632ef6dcce6a7dfec9a229c7be4d6d5c894/packages/remote-server/src/web-did-doc-router.ts#L40)). We should make this assumption more clear there too.

To get to that point we need to use [one of the ENS SDKs](https://docs.ens.domains/dapp-developer-guide/ens-libraries#javascript) to modify the actual text records mentioned in the [did:ens spec](https://github.com/veramolabs/did-ens-spec).

Reply:
That's a good point, although this kind of race-condition is probably true of other DID methods related to ethereum as well.